modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ObjectFactoryClassRepository.java

```diff
@@ -125,6 +129,10 @@
           return proxy.invokeSuper(obj, args);
         }
     });
+
+    if (originalClassLoader != null) {
```

Comment:
If the previous code throws an exception this code is never call. It must so it should be in a finally clause